src/main/java/com/powsybl/caseserver/service/FsCaseService.java

```diff
@@ -100,20 +100,25 @@ String getFormat(Path caseFile) {
     public List<CaseInfos> getCases() {
         try (Stream<Path> walk = Files.walk(getStorageRootDir())) {
             return walk.filter(Files::isRegularFile)
-                    .map(this::getCaseInfos)
+                    .map(this::getCaseInfosOrNull)
                     .filter(Objects::nonNull)
-                    .map(this::removeGzipExtensionFromPlainFile)
                     .toList();
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
     private CaseInfos getCaseInfos(Path file) {
+        return createInfos(file, UUID.fromString(file.getParent().getFileName().toString()));
+    }
+
+    private CaseInfos getCaseInfosOrNull(Path file) {
         Objects.requireNonNull(file);
         try {
-            return createInfos(file, UUID.fromString(file.getParent().getFileName().toString()));
+            CaseInfos caseInfos = getCaseInfos(file);
+            return Objects.nonNull(caseInfos) ? removeGzipExtensionFromPlainFile(caseInfos) : null;
```

Comment:
given the code in removeGzipExtensionFromPlainFile
we could simplify to 
```suggestion
            return removeGzipExtensionFromPlainFile(caseInfos);
```

Reply:
done